blog/2021/02/julia-1.6-highlights.md

```diff
@@ -0,0 +1,335 @@
+result is a leaner, faster Julia that is far more impervious to method
+invalidation, and feels considerably more responsive and nimble in interactive
+sessions. Related blog post: <https://julialang.org/blog/2020/08/invalidations/>.
+
+## Compiler latency reduction
+
+_Jeff Bezanson_
+
+## Tooling to help optimize packages for latency
+
+_Nathan Daly & Tim Holy_
+
+Julia 1.6, in conjunction with SnoopCompile v2.2.0 or higher, features new tools
+for compiler introspection, especially (but not exclusively) for type inference.
+Developers can use the new tools to profile type inference and determine how
+particular package implementation choices interact with compilation time. Early
+adopters have used these tools to eliminate anywhere from a few percent to the
+large majority of first-use latency.
+
+- Related blog post: <https://julialang.org/blog/2021/01/precompile_tutorial/>
+- Documentation: <https://timholy.github.io/SnoopCompile.jl/stable/>
+
+## Binary loading speedups
+
+_Elliot Saba & Mosè Giordano_
+
+Binary packages have been a problematic part of the Julia world for its entire
+existence.  First, installation troubles plagued the ecosystem and conflicts
+due to system differences were common.  We addressed these issues through ever-
+increasingly isolated installation processes until we converged to the current
+standard installation procedure; namely installation of binaries as artifacts
+typically sourced from a [`BinaryBuilder.jl`](https://github.com/JuliaPackaging/BinaryBuilder.jl) cross-compilation recipe.  Libraries built from `BinaryBuilder.jl` are most
+often used through so-called JLL packages which provide a standardized API that
+Julia packages can use to access the provided binaries.  This ease of use and
+reliability of installation came at a cost, however, which was _vastly_
+increased load times as compared to the bad old days when Julia packages would
+blindly `dlopen()` libraries and load whatever libraries happened to be sitting
+on the library search path.  To illustrate the issue, in Julia 1.4, loading the
+GTK+3 stack required **8 seconds** when it used to take around **500ms** on the
+same machine.  Through many months of hard work and careful investigation, we
+are pleased to report that the same stack of libraries now takes less than
+**200ms** to load on Julia v1.6.
+
+The cause of this slowdown was multi-faceted and spread across many different
+layers of the Julia ecosystem.  Part of the issue was general compiler latency,
+which has been a focus of the compiler team for some time now, as evidenced by
+Jeff's post in this blogpost.  Another major piece though was general overhead
+incurred by having so many small JLL packages providing bindings; there was
+significant overhead in the loading of each package.  In particular, there was
+code inferrence, code generation and datastructure loading that needed to be
+eliminated if the JLL packages were to be lightweight enough to not effect
```

Comment:
```suggestion
eliminated if the JLL packages were to be lightweight enough to not affect
```

Reply:
Although I know the difference, I apparently don't know it well enough for the knowledge to effect me.